src/main/scala/io/iohk/ethereum/network/TimeSlotStats.scala

```diff
@@ -13,8 +13,10 @@ class TimeSlotStats[K, V: Monoid] private (
     // The last written position in the buffer.
     val lastIdx: Int,
     // Ring buffer of slots statistics.
-    val buffer: TimeSlotStats.Buffer[K, V]
-)(implicit clock: Clock) {
+    val buffer: TimeSlotStats.Buffer[K, V],
+    // The time counter counter
+    clock: Clock
+) {
```

Comment:
What was the problem with the way it was? I used an implicit just to make it a bit easier to return new instances of this otherwise immutable class.

Reply:
It also seemed to make some sense that different parts of the app shouldn't have different clocks, so passing around an implicit one to whoever needs it might be better than threading it through. But ultimately it was only passed on in once place in this class so I get it if you don't like it.

Reply:
Made clock immutable again